src/lib/tuning/index/index_evaluator.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <map>
+#include <memory>
+#include <set>
+#include <utility>
+
+#include "sql/sql_query_cache.hpp"
+#include "sql/sql_query_plan.hpp"
+#include "tuning/index/base_index_evaluator.hpp"
+#include "tuning/index/column_ref.hpp"
+#include "tuning/index/index_choice.hpp"
+
+namespace opossum {
+
+/**
+ * This is the default implementation of a BaseIndexEvaluator.
+ *
+ * It only considers single-column indexes for now.
+ *
+ * Desirability is determined by the amount of "saved work" in terms on
+ * "unscanned rows", i.e. the number of rows that don't have to be read when
+ * using an index on a given column compared to a linear table scan.
+ * This assumes that columns are always scanned completely, meaning that
```

Comment:
Couldn't we do this? The LQP should give us information about previously scanned columns

Reply:
True. We also thought about this, but since you told us repeatedly to not get too fancy with our index evaluator as it should only work as an example, we didn't continue down that road and just documented the underlying assumption / limitation of the implemented algorithm.